src/test/java/core/basesyntax/MainTest.java

```diff
@@ -0,0 +1,134 @@
+package core.basesyntax;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class MainTest {
+    private static final String MESSAGE_NPE = "NPE was expected!";
+    private static final String MESSAGE_POW =
+            "This calculator can't work with negative pow or with very big range!";
+
+    @Test
+    public void correctSum() {
+        Assert.assertEquals("3", Main.calculating(Main::lambdaImpl, -2, '+', 5));
```

Comment:
```suggestion
        Assert.assertEquals("3", Main.calculating(Calculator::lambdaImpl, -2, '+', 5));
``` 

Може краще викликати не з класу Main а з якого класу Calculator?

Reply:
Ну, тут більшу половину всього тре перероблювати :-)  бо з умови задачі цього не ясно.